compiler/rustc_hir_typeck/src/fn_ctxt/checks.rs

```diff
@@ -1589,26 +1589,70 @@ impl<'a, 'tcx> FnCtxt<'a, 'tcx> {
                 // e.g. `reuse HasSelf::method;` should suggest `reuse HasSelf::method($args);`.
                 full_call_span.shrink_to_hi()
             };
+
+            // Controls how the arguments should be listed in the suggestion.
+            enum ArgumentsFormatting {
+                SingleLine,
+                Multiline { fallback_indent: String, brace_indent: String },
+            }
+            let arguments_formatting = {
+                let mut provided_inputs = matched_inputs.iter().filter_map(|a| *a);
+                if let Some(brace_indent) = source_map.indentation_before(suggestion_span)
+                    && let Some(first_idx) = provided_inputs.by_ref().next()
+                    && let Some(last_idx) = provided_inputs.by_ref().next()
+                    && let (_, first_span) = provided_arg_tys[first_idx]
+                    && let (_, last_span) = provided_arg_tys[last_idx]
+                    && source_map.is_multiline(first_span.to(last_span))
+                    && let Some(fallback_indent) = source_map.indentation_before(first_span)
+                {
+                    ArgumentsFormatting::Multiline { fallback_indent, brace_indent }
+                } else {
+                    ArgumentsFormatting::SingleLine
+                }
+            };
+
             let mut suggestion = "(".to_owned();
             let mut needs_comma = false;
             for (expected_idx, provided_idx) in matched_inputs.iter_enumerated() {
-                if needs_comma {
-                    suggestion += ", ";
-                } else {
-                    needs_comma = true;
+                match (needs_comma, &arguments_formatting) {
+                    (true, ArgumentsFormatting::SingleLine) => {
+                        suggestion += ", ";
+                    }
+                    (false, ArgumentsFormatting::SingleLine) => {
+                        needs_comma = true;
+                    }
+                    (true, ArgumentsFormatting::Multiline { .. }) => {
+                        suggestion += ",\n";
+                    }
+                    (false, ArgumentsFormatting::Multiline { .. }) => {
+                        suggestion += "\n";
+                        needs_comma = true;
+                    }
```

Comment:
This seems separable. Why not just:

`if needs_comma { suggestion += ", "; } else { needs_comma = true; }` 

then match on `ArgumentsFormatting` and append `"\n"` if it's multiline?